platform/sensinact-shell/src/main/java/org/eclipse/sensinact/gateway/commands/gogo/SensiNactCommandSession.java

```diff
@@ -0,0 +1,61 @@
+/*********************************************************************
+* Copyright (c) 2024 Contributors to the Eclipse Foundation.
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which is available at https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*   Kentyou - initial implementation
+**********************************************************************/
+package org.eclipse.sensinact.gateway.commands.gogo;
+
+import org.eclipse.sensinact.core.push.DataUpdate;
+import org.eclipse.sensinact.core.session.SensiNactSession;
+import org.eclipse.sensinact.core.session.SensiNactSessionManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.util.promise.Promise;
+
+/**
+ * We want to reuse the same session for all of the commands,
+ * so it is easier to have a single reference to this class, rather than
+ * trying to coordinate access to the same service from each of the
+ * separate components in this bundle.
+ *
+ * @author David Leangen
+ */
+@Component(service = SensiNactCommandSession.class)
+public class SensiNactCommandSession {
+
+    @Reference private SensiNactSessionManager sessionManager;
+    private SensiNactSession session;
+
+    // Not technically a session, but still a good place to keep this reference
+    @Reference private DataUpdate push;
+
+    @Activate
+    void activate() {
+        session = sessionManager.createNewAnonymousSession();
+    }
+
+    /**
+     * Returns the command session. Value will never be null;
+     */
+    public SensiNactSession get() {
+        // TODO: Can the session expire?
```

Comment:
Yes, sessions can and do expire

https://github.com/eclipse/org.eclipse.sensinact.gateway/blob/969d78788b379e093a51fea516e8933c8710674e/core/api/src/main/java/org/eclipse/sensinact/core/session/SensiNactSession.java#L39-L61

Reply:
Thanks! Any advice as to how long to extend for? I was thinking an hour...